general/chainerrl/baselines/gail.py

```diff
@@ -0,0 +1,786 @@
+"""
+MIT License
+
+Copyright (c) Preferred Networks, Inc.
+"""
+
+import argparse
+from logging import getLogger
+import os
+
+import minerl  # noqa: register MineRL envs as Gym envs.
+import gym
+import numpy as np
+
+import chainer
+from chainer import functions as F
+from chainer import links as L
+
+import chainerrl
+from chainerrl import experiments
+from chainerrl.wrappers.atari_wrappers import (
+    FrameStack, ScaledFloatFrame, LazyFrames)
+from agents.behavioral_cloning import BehavioralCloning
+from agents.gail import Discriminator, GAIL
+from expert_dataset import ExpertDataset
+from observation_wrappers import (
+    generate_pov_converter, generate_pov_with_compass_converter,
+    generate_unified_observation_converter)
+from action_wrappers import (
+    generate_discrete_converter, generate_continuous_converter,
+    generate_multi_dimensional_softmax_converter)
+from distribution import MultiDimensionalSoftmaxDistribution
+from utils import ordinal_logit_function
+
+from env_wrappers import (
+    SerialDiscreteActionWrapper, NormalizedContinuousActionWrapper,
+    UnifiedObservationWrapper, GrayScaleWrapper, PoVWithCompassAngleWrapper,
+    ObtainPoVWrapper, FrameSkip, MoveAxisWrapper,
+    MultiDimensionalSoftmaxActionWrapper)
+
+logger = getLogger(__name__)
+
+
+class ActorTRPONetForDiscrete(chainer.Chain):
```

Comment:
Maybe move the policies to a separate policies.py file, like the q_functions.py

Reply:
I understand. I moved them to `policies.py`. Thank you!